SingularityUI/app/components/common/Table.cjsx

```diff
@@ -0,0 +1,47 @@
+Utils = require '../../utils'
+
+Table = React.createClass
+
+    renderTableHeader: ->
+        header = []
+        key = 0
+        for columnName in @props.columnNames
+            header.push(<th key={key}>{columnName}</th>)
+            key++
+        return header
+
+    renderTableRow: (elements) ->
+        data = []
+        key = 0
+        for element in elements
+            data.push(<td key={key}>{element}</td>)
+            key++
+        return data
+
+    # Note: The data in @props.tableRows should be the JSX you would like to render
+    renderTableData: ->
+        tableRows = []
+        key = 0
+        for tableRow in @props.tableRows
+            tableRows.push(<tr key={key}>{@renderTableRow tableRow}</tr>)
+            key++
+        return tableRows
+
+
+    # Note: Use @props.tableClassOpts to declare things like striped or bordered
+    getClassName: ->
+        return "table " + @props.tableClassOpts
```

Comment:
There's a handy `classnames` function in [react-utils](https://git.hubteam.com/HubSpot/react-utils). I highly recommend using it instead of string concatenation, and using a `className` prop instead of `tableClassOpts`. So this would become:

```coffee
getClassName: ->
  classNames('table', @props.className)
```